dm/master/server.go

```diff
@@ -217,7 +217,11 @@ func (s *Server) recoverSubTask() error {
 		return err
 	}
 	for _, kv := range resp.Kvs {
-		infos := common.UpstreamSubTaskKeyAdapter.Decode(string(kv.Key))
+		infos, err := common.UpstreamSubTaskKeyAdapter.Decode(string(kv.Key))
+		if err != nil {
+			log.L().Warn("decode upstream subtask key from etcd failed", zap.Error(err))
+			continue
```

Comment:
How about returning the `err` directly? I think it's better to send the error to the user  ASAP.